### PR TITLE
best effort testing of SignalingClient ✅ 🚀

### PR DESCRIPTION
**Pull request by bengreenier**
## Checklist
***This checklist is used to make sure that common guidelines for a pull request are followed.***

- [x] **I have read the [contribution guidelines](../blob/master/CONTRIBUTING.md).**
- [x] Title of the pull request is clear and informative.
- [ ] If applicable, the code is properly documented.
- [ ] The code builds without any errors.
- [ ] All unit and integration tests pass.

## Description

 + `authentication_provider` public api tests
 + `peer_connection_client` public api tests
 + `ssl_capable_socket` public api tests
 + `turn_credential_provider` public api tests

As documented/requested in #185 

## Related Issues and PRs

+ #185

### CODE REVIEW

**Comment by bengreenier** [comment]
i'm still seeing transient vcpkg runs in CI. I don't know why that would be occuring...

**Comment by bengreenier** [comment]
looks like there's still some vcpkg work that CANNOT be done in parallel. I'll refactor so only the vcpkg project runs vcpkg 😓

**Comment by bengreenier** [comment]
this "right way ™️" to solve this was to make the msbuild wrapper block on multiple calls. i've published that downlevel change and will validate it here (it works in unit tests/small manual test case) once it is live on nuget.org - see https://github.com/bengreenier/vcpkg.nuget/releases/tag/v1.3.0

**Comment by bengreenier** [comment]
if this is green (i believe it should be) then we are finally good to go here 😓 🙇

File: Plugins/UnityServerPlugin/StreamingUnityServerPlugin.cpp [hunk 1]
```diff
@@ -155,6 +155,11 @@ struct UnityServerPeerObserver : public PeerConnectionClientObserver,
 		}
 	}
 
+	virtual void OnHeartbeat(int heartbeat_status) override
+	{
+		// TODO(bengreenier): wire up to unity
```
Comment (anderm) [inline Plugins/UnityServerPlugin/StreamingUnityServerPlugin.cpp]: Can you create an issue for this?

File: Samples/Client/DirectxWin32/src/conductor.cpp [hunk 2]
```diff
@@ -470,6 +470,11 @@ void Conductor::OnMessageSent(int err)
 	main_window_->QueueUIThreadCallback(SEND_MESSAGE_TO_PEER, NULL);
 }
 
+void Conductor::OnHeartbeat(int heartbeat_status)
+{
+	// TODO(bengreenier): if we want to surface this so it can be observed, we could do so
```
Comment (anderm) [inline Samples/Client/DirectxWin32/src/conductor.cpp]: Same here

File: Plugins/UnityServerPlugin/StreamingUnityServerPlugin.cpp (hunk 1 above)
Comment (bengreenier) [reply Plugins/UnityServerPlugin/StreamingUnityServerPlugin.cpp]: see https://github.com/CatalystCode/3DStreamingToolkit/issues/259

File: Samples/Client/DirectxWin32/src/conductor.cpp (hunk 2 above)
Comment (bengreenier) [reply Samples/Client/DirectxWin32/src/conductor.cpp]: see https://github.com/CatalystCode/3DStreamingToolkit/issues/260

**Pull request merged**